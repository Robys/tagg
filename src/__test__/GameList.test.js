import React from 'react'
import {act} from 'react-dom/test-utils'
import {shallow} from '../enzyme'
import GameList from '../components/GamesList'

const whenStable = async () =>
  await act(async () => {
    await new Promise((resolve) => setTimeout(resolve, 0));
  });


  describe("testing <GameList/>",()=>{

    test("fetching game list", async ()=>{
        const container = shallow(<GameList />)

        await whenStable()
        expect(container.find("#games-list")).toBeTruthy()
        expect(container).toMatchSnapshot()

    })

})