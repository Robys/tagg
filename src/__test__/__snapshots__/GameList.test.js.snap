// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <GameList/> fetching game list 1`] = `
<WithStyles(ForwardRef(Card))
  className="game-list-card"
>
  <WithStyles(ForwardRef(Paper))
    component="form"
    style={
      Object {
        "backgroundColor": "#FD507A",
        "color": "white",
        "height": "100px",
      }
    }
  >
    <WithStyles(ForwardRef(InputBase))
      className="game-list-input"
      inputProps={
        Object {
          "aria-label": "procure aqui",
        }
      }
      onChange={[Function]}
      placeholder="Procure aqui"
    />
    <WithStyles(ForwardRef(IconButton))
      aria-label="search"
      onClick={[Function]}
    >
      <Memo(ForwardRef(SearchIcon)) />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(List))
    style={
      Object {
        "maxHeight": "100%",
        "overflow": "auto",
      }
    }
  >
    <div>
      <WithStyles(ForwardRef(Skeleton))
        animation="wave"
        height={380}
        variant="rect"
        width={380}
      />
      <WithStyles(ForwardRef(Skeleton))
        animation="wave"
        variant="text"
      />
    </div>
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Card))>
`;
